chains/evm/contracts/pools/ERC20LockBox.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {TokenAdminRegistry} from "../tokenAdminRegistry/TokenAdminRegistry.sol";
+
+import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
+
+import {IERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @title ERC20 Lock Box
+/// @notice A contract that holds ERC20 tokens for a token pool to simplify pool upgrades without requiring a liquidity migration.
+contract ERC20LockBox is Ownable2StepMsgSender {
+  using SafeERC20 for IERC20;
+
+  error Unauthorized(address caller);
+  error InsufficientBalance(uint64 remoteChainSelector, uint256 requested, uint256 available);
+  error TokenAmountCannotBeZero();
+  error RecipientCannotBeZeroAddress();
+  error TokenAddressCannotBeZero();
+  error TokenAdminRegistryCannotBeZeroAddress();
+
+  event AllowedCallerAdded(address indexed token, address indexed caller);
+  event AllowedCallerRemoved(address indexed token, address indexed caller);
+  event Deposit(address indexed token, uint64 indexed remoteChainSelector, address indexed depositor, uint256 amount);
+  event Withdrawal(
+    address indexed token, uint64 indexed remoteChainSelector, address indexed recipient, uint256 amount
+  );
+
+  struct AllowedCallerConfigArgs {
+    address token;
+    address caller;
+    bool allowed;
+  }
+
+  TokenAdminRegistry public immutable i_tokenAdminRegistry;
+
+  mapping(address => mapping(address => bool)) internal s_allowedCallers;
+  mapping(address => mapping(uint64 => uint256)) internal s_tokenBalances;
+
+  constructor(
+    address tokenAdminRegistry
+  ) {
+    if (tokenAdminRegistry == address(0)) {
+      revert TokenAdminRegistryCannotBeZeroAddress();
+    }
+    i_tokenAdminRegistry = TokenAdminRegistry(tokenAdminRegistry);
+  }
+
+  /// @notice Deposits tokens for a specific remote chain selector
```

Comment:
This comment explains *what* happens, not *why*. Why would anyone use this function?